openedx/core/djangoapps/plugins/plugin_contexts.py

```diff
@@ -0,0 +1,81 @@
+from importlib import import_module
+
+from logging import getLogger
+
+from openedx.core.lib.cache_utils import process_cached
+
+from . import constants, registry
+
+
+log = getLogger(__name__)
+
+def get_plugins_view_context(project_type, view_name, existing_context={}):
+    """
+    Returns a dict of additonal view context. Will check if any plugin apps
+    have that view in their view_context_config, and if so will call their
+    selected function to get their context dicts.
+    """
+    aggregate_context = {"plugins": {}}
+
+    # This functionality is cached
+    context_functions = _get_context_functions_for_view(project_type, view_name)
+
+    for (context_function, plugin_name) in context_functions:
+        plugin_context = context_function(existing_context)
+        try:
+            plugin_context = context_function(existing_context)
+        except Exception as exc:
+            # We're catching this because we don't want the core to blow up when a
+            # plugin is broken. This exception will probably need some sort of
+            # monitoring hooked up to it to make sure that these errors don't go
+            # unseen.
+            log.exception("Failed to call plugin context function. Error: %s", exc)
+            continue
+
+        aggregate_context["plugins"][plugin_name] = plugin_context
+
+    return aggregate_context
+
+
+def _get_context_function(app_config, project_type, view_name):
```

Comment:
Nit: Move `_get_context_function` below `_get_context_functions_for_view` so your code reads top-down in this file (as recommended in the Clean Code book).  That is, `_get_context_functions_for_view` is a higher level function above `_get_context_function` and so comes before the lower level helpers.